code/hmc/test_hmc.py

```diff
@@ -1,7 +1,15 @@
+
+from __future__ import print_function
+
 import numpy
 import theano
 
-from hmc import HMC_sampler
+try:
+    from hmc import HMC_sampler
+except:
+    # python 3 compatibility
+    # http://stackoverflow.com/questions/3073259/python-nose-import-error
+    from hmc.hmc import HMC_sampler
```

Comment:
So I have more insight about this.
In this case, the issue is a clash of names between the `hmc/` directory and the `hmc.py` file inside. On Python 3, `nosetests` thinks `hmc` is the directory, but the regular Python interpreter thinks it is `hmc.py` (if you try to import `test_hmc` for instance, or run `python test_hmc.py`).
This looks like an issue with nosetests to me. Possible solutions would include:
- Yours (except that I would catch only `ImportError` rather than anything)
- Having different names, by either renaming `hmc/` or `hmc.py`. That would break direct links, though.

I would prefer the first one.
However, for `midi`, I'm not sure that the issue is the same exactly.